src/Place/ReadModel/RDF/RdfProjector.php

```diff
@@ -0,0 +1,118 @@
+        GraphRepository $graphRepository,
+        IriGeneratorInterface $iriGenerator
+    ) {
+        $this->mainLanguageRepository = $mainLanguageRepository;
+        $this->graphRepository = $graphRepository;
+        $this->iriGenerator = $iriGenerator;
+    }
+
+    public function handle(DomainMessage $domainMessage): void
+    {
+        $payload = $domainMessage->getPayload();
+        $granularEvents = $payload instanceof ConvertsToGranularEvents ? $payload->toGranularEvents() : [];
+        $events = [$payload, ...$granularEvents];
+
+        $uri = $this->iriGenerator->iri($domainMessage->getId());
+
+        $mapping = [
+            MainLanguageDefined::class => fn ($e) => $this->handleMainLanguageDefined($e, $uri),
+            TitleUpdated::class => fn ($e) => $this->handleTitleUpdated($e, $uri),
+            TitleTranslated::class => fn ($e) => $this->handleTitleTranslated($e, $uri),
+        ];
+
+        foreach ($events as $event) {
+            foreach ($mapping as $class => $handler) {
+                if ($event instanceof $class) {
+                    $handler($event);
+                }
+            }
+        }
+    }
+
+    private function handleMainLanguageDefined(MainLanguageDefined $event, string $uri): void
+    {
+        $this->mainLanguageRepository->save($uri, new Language($event->getMainLanguage()->getCode()));
+    }
+
+    private function handleTitleUpdated(TitleUpdated $event, string $uri): void
+    {
+        $graph = $this->graphRepository->get($uri);
+        $mainLanguage = $this->mainLanguageRepository->get($uri, new Language('nl'));
+
+        $resource = $graph->resource($uri);
+
+        $this->replaceLanguageValue(
+            $resource,
+            self::PROPERTY_LOCATIE_NAAM,
+            $event->getTitle()->toNative(),
+            $mainLanguage->toString(),
+        );
+
+        $this->graphRepository->save($uri, $graph);
+    }
+
+    private function handleTitleTranslated(TitleTranslated $event, string $uri): void
+    {
+        $graph = $this->graphRepository->get($uri);
+
+        $resource = $graph->resource($uri);
+
+        $this->replaceLanguageValue(
+            $resource,
+            self::PROPERTY_LOCATIE_NAAM,
+            $event->getTitle()->toNative(),
+            $event->getLanguage()->getCode()
+        );
+
+        $this->graphRepository->save($uri, $graph);
+    }
+
+    private function replaceLanguageValue(
+        Resource $resource,
+        string $property,
+        string $value,
+        string $language
+    ): void {
+        /** @var Literal $literal */
+        foreach ($resource->allLiterals($property) as $literal) {
+            if ($literal->getLang() === $language) {
+                $resource->delete($property, $literal);
+            }
+        }
+
+        $resource->add($property, [
+            'type' => 'literal',
+            'value' => $value,
+            'lang' => $language,
+        ]);
```

Comment:
```suggestion
        $resource->add(
                $property,
                new Literal($value, $language).toArray()
        );
```

Maybe a literal could be created with its own class?

Reply:
Apparently the `easyrdf/easyrdf` library already has a `Literal` class which is supported in this method, so I've used that instead.